lib/urlwatch/config.py

```diff
@@ -0,0 +1,114 @@
+
+class CommandConfig(BaseConfig):
+
+    def __init__(self, pkgname, urlwatch_dir, bindir, prefix, config, urls, hooks, cache, verbose):
+
+        super().__init__(pkgname, urlwatch_dir, bindir, prefix, config, urls, cache, hooks, verbose)
+
+        if self.bindir == 'bin':
+            # Installed system-wide
+            self.examples_dir = os.path.join(prefix, 'share', self.pkgname, 'examples')
+        else:
+            # Assume we are not yet installed
+            self.examples_dir = os.path.join(prefix, bindir, 'share', self.pkgname, 'examples')
+
+        self.urls_yaml_example = os.path.join(self.examples_dir, 'urls.yaml.example')
+        self.hooks_py_example = os.path.join(self.examples_dir, 'hooks.py.example')
+
+        self.parse_args()
+
+    def parse_args(self):
+
+        parser = argparse.ArgumentParser(description=urlwatch.__doc__,
+                                         formatter_class=argparse.RawDescriptionHelpFormatter)
+        parser.add_argument('--version', action='version', version='%(prog)s {}'.format(urlwatch.__version__))
+        parser.add_argument('-v', '--verbose', action='store_true', help='show debug output')
+        group = parser.add_argument_group('files and directories')
+        group.add_argument('--urls', metavar='FILE', help='read job list (URLs) from FILE',
+                           default=self.urls)
+        group.add_argument('--config', metavar='FILE', help='read configuration from FILE',
+                           default=self.config)
+        group.add_argument('--hooks', metavar='FILE', help='use FILE as hooks.py module',
+                           default=self.hooks)
+        group.add_argument('--cache', metavar='FILE', help='use FILE as cache database',
+                           default=self.cache)
+
+        group = parser.add_argument_group('job list management')
+        group.add_argument('--list', action='store_true', help='list jobs')
+        group.add_argument('--add', metavar='JOB', help='add job (key1=value1,key2=value2,...)')
+        group.add_argument('--delete', metavar='JOB', help='delete job by location or index')
+        group = parser.add_argument_group('interactive commands ($EDITOR/$VISUAL)')
+        group.add_argument('--edit', action='store_true', help='edit URL/job list')
+        group.add_argument('--edit-config', action='store_true', help='edit configuration file')
+        group.add_argument('--edit-hooks', action='store_true', help='edit hooks script')
+        group = parser.add_argument_group('miscellaneous')
+        group.add_argument('--features', action='store_true', help='list supported jobs/filters/reporters')
+        group.add_argument('--gc-cache', action='store_true', help='remove old cache entries')
+
+        args = parser.parse_args()
+
+        i = 0
+        for arg in vars(args):
+
+            argval = getattr(args, arg)
+
+            # print("Arg #{0}: {1} = {2}".format(str(i),arg,str(argval)))
```

Comment:
Remove commented-out code that is not used. Also note PEP-8 here (too many whitespace lines where there should be none. Also note that instead of keeping track of the `i` variable manually, you can do:

```
for i, arg in enumerate(vars(args)):
    ...
```